demos/connector/neo4j/load-cora-into-neo4j.ipynb

```diff
@@ -0,0 +1,413 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import py2neo\n",
+    "\n",
+    "# Create the Neo4J Graph database object\n",
+    "graph = py2neo.Graph(\"http://localhost:7474\", username=\"neo4j\", password=\"cora\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Delete the existing edges and relationships in the current database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "empty_db_query = \"\"\"\n",
+    "    MATCH(n) DETACH\n",
+    "    DELETE(n)\n",
+    "    \"\"\"\n",
+    "\n",
+    "tx = graph.begin(autocommit=True)\n",
+    "tx.evaluate(empty_db_query)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load all nodes to the graph database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "loading_node_query = \"\"\"\n",
+    "    UNWIND $node_list as node\n",
+    "    CREATE( e: paper {\n",
+    "        ID: toInteger(node.id),\n",
+    "        subject: node.subject,\n",
+    "        features: node.features\n",
+    "    })\n",
+    "    \"\"\"\n",
+    "\n",
+    "# For efficient loading, we will load batch of nodes into Neo4J.\n",
+    "batch_len = 500\n",
+    "\n",
+    "for batch_start in range(0, len(node_list), batch_len):\n",
+    "    batch_end = min(len(node_list), batch_start + batch_len)\n",
+    "    # turn node dataframe into a list of records\n",
+    "    records = node_list.iloc[batch_start:batch_end].to_dict(\"records\")\n",
+    "    tx = graph.begin(autocommit=True)\n",
+    "    tx.evaluate(loading_node_query, parameters={\"node_list\": records})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load all edges to the graph database."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "loading_edge_query = \"\"\"\n",
+    "    UNWIND $edge_list as edge\n",
+    "    \n",
+    "    MATCH(source: paper {ID: toInteger(edge.source)})\n",
+    "    MATCH(target: paper {ID: toInteger(edge.target)})\n",
+    "    \n",
+    "    MERGE (source)-[r:cites]->(target)\n",
+    "    \"\"\"\n",
+    "\n",
+    "batch_len = 500\n",
+    "\n",
+    "for batch_start in range(0, len(edge_list), batch_len):\n",
+    "    batch_end = min(len(edge_list), batch_start + batch_len)\n",
+    "    # turn edge dataframe into a list of records\n",
+    "    records = edge_list.iloc[batch_start:batch_end].to_dict(\"records\")\n",
+    "    tx = graph.begin(autocommit=True)\n",
+    "    tx.evaluate(loading_edge_query, parameters={\"edge_list\": records})"
```

Comment:
How long does this take to run? I vaguely recall people saying that Neo4J is really slow to load data directly, and using CSVs is fastest.

Reply:
Yeah you're right! 
Cora dataset is tiny, so in my machine, it takes:
- 2.9 s to load all the nodes, 
- 12.5 s to load all the edges.

For bigger dataset, it is recommended to use ```neo4j admin``` tool. This loads the dataset directly from the CSVs.